### ISSUE TITLE
Parallel requests causes losing some stdout

### ISSUE THREAD

**Issue by Brikaa** [engineer-man/piston#500]
![image](https://user-images.githubusercontent.com/40431124/177150078-86a224e8-9876-45fd-84c7-d651007c3348.png)

Adding a 0 ms sleep here seems to fix it for some reason:
```js
proc.on('exit', async (code, signal) => {
    await new Promise(resolve => set_timeout(resolve, 0));
    this.exit_cleanup();

    resolve({ stdout, stderr, code, signal, output });
});
```

Commenting the cleanup doesn't fix it

**Comment by gouseferoz** [comment]
Hi, I am also facing this issue.  Any solutions on how to fix it?

**Comment by Brikaa** [comment]
Fixed with the backport PR #613
